onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -942,6 +943,7 @@ REGISTER_KERNEL_TYPED_12(ReduceMax, MLFloat16)
 REGISTER_KERNEL_TYPED_12(ReduceMax, float)
 REGISTER_KERNEL_TYPED_12(ReduceMax, double)
 REGISTER_KERNEL_TYPED_12(ReduceMax, int32_t)
+REGISTER_KERNEL_TYPED_12(ReduceMax, int64_t)
```

Comment:
There are tests for Reduce* already for int64 type (which was previousky running only on CPU - now they will get tested on CUDA)